### ISSUE TITLE
tagNamePreference tag value not handled properly

### ISSUE THREAD

**Issue by markdoten** [gajus/eslint-plugin-jsdoc#435]
I'm trying to disable the use of the `@description` tag for jsdoc descriptions. E.g.
```
/**
 * Some description
 */
```
Instead of:
```
/**
 * @description Some description
 */
```

My settings are:
```
settings: {
    jsdoc: {
      tagNamePreference: {
        description: false
      }
    }
},
rules: {
  'jsdoc/require-description': 2
}
```
When I run the linter, it gives me this error:
```
error  Missing JSDoc @false declaration  jsdoc/require-description
```
Based on the [documentation](https://github.com/gajus/eslint-plugin-jsdoc#alias-preference), I would expect providing `false` to `reject a normally valid tag`.

Am I just not understanding how it's supposed to work? Am I missing a necessary rule?

Versions:
- eslint: `^6.6.0`
- eslint-plugin-jsdoc: `^18.1.2`

**Comment by brettz9** [comment]
Based on the error you have reported, it looks like you must be supplying a string "false":

```js
    settings: {
        jsdoc: {
          tagNamePreference: {
            description: 'false',
          },
        },
      },
```

...whereas it should be a boolean:

```js
    settings: {
        jsdoc: {
          tagNamePreference: {
            description: false,
          },
        },
      },
```

Also based on the error you reported, it appears you must be using the option `descriptionStyle` set to "tag", whereas it appears you want to set it to "body", ensuring that the `require-description` rule does not try to check for the `@description` tag.

You will get an error if you try to set `descriptionStyle` to "tag" and then use `tagNamePreference: {description: false}` with a `@description`, because it means you have opted to allow for checking the `@description` tag for `require-description` (by `descriptionStyle: 'tag'`) yet you have also prevented this tag generically (by `tagNamePreference: {description: false}`).

Normally, though, if you want to prevent `@description` anywhere, you should instead use the rule `check-tag-names` and that will check `settings.jsdoc.tagNamePreference` to determine that `@description` should be reported.

Closing as I think that should address, though feel free to comment further if this does not address.

**Comment by markdoten** [comment]
@brettz9 `check-tag-names` was what I was looking for, thanks!
